roster2/tool/src/java/org/sakaiproject/roster/tool/entityprovider/RosterPOIEntityProvider.java

```diff
@@ -478,22 +478,22 @@ private void addOverviewRows(final List<List<String>> dataInRows,
 				row.add(member.getEmail());
 			}
 
-			if (this.sakaiProxy.getViewCandidateDetails(siteId)) {
-				row.add(member.getStudentNumber());
-			}
-
 			if (this.sakaiProxy.getViewUserProperty(siteId)) {
 				List<String> props = member.getUserProperties().entrySet().stream().map(e -> e.getKey() + ":" + e.getValue()).collect(Collectors.toList());
 				row.add(String.join("\n", props));
 			}
 
 			if (this.sakaiProxy.getViewCandidateDetails(siteId)) {
-				List<String> specialNeeds = member.getSpecialNeeds();
-				row.add(specialNeeds != null ? String.join("\n", specialNeeds) : "");
+				List<String> additionalNotes = member.getAdditionalNotes();
+				row.add(additionalNotes != null ? String.join("\n", additionalNotes) : "");
 			}
 
 			if (this.sakaiProxy.getViewCandidateDetails(siteId)) {
-				List<String> specialNeeds = member.getAdditionalNotes();
+				row.add(member.getStudentNumber());
+			}
+
+			if (this.sakaiProxy.getViewCandidateDetails(siteId)) {
+				List<String> specialNeeds = member.getSpecialNeeds();
```

Comment:
Couldn't all these 3 conditions go under the same if?

Reply:
Yes!

Reply:
Yes!